opal-ui/src/stores/users.ts

```diff
@@ -0,0 +1,45 @@
+import { defineStore } from 'pinia';
+import { api } from 'src/boot/api';
+import { SubjectCredentialsDto } from 'src/models/Opal';
+
+export const useUsersStore = defineStore('users', () => {
+  const users = ref([] as SubjectCredentialsDto[]);
+
+  function reset() {
+    users.value = [];
+  }
+
+  async function initUsers() {
+    reset();
+    await loadUsers();
+  }
+
+  async function loadUsers() {
+    const response = await api.get('/system/subject-credentials');
+    users.value = response.data;
+  }
+
+  async function addUser(user: SubjectCredentialsDto) {
+    await api.post(`/system/subject-credentials`, user);
```

Comment:
The other stores return a Promise instead, because is it up to the store's client to decide to await or not, for instance, there is no reason for blocking on loadUsers, let the reactive magic happen.